doc/simd_api.rst

```diff
@@ -719,22 +801,46 @@ of type *J*, used only to disambiguate overloads.
       - Type
       - Description
 
-    * - ``C::gather(J{}, p, j)``
+    * - ``C::gather(tag<J>{}, p, j)``
       - ``C::vector_type``
       - Vector *v* with values *v*\ `i`:sub: = ``p[j[i]]``.
 
-    * - ``C::gather(J{}, u, p, j, m)``
+    * - ``C::gather(tag<J>{}, u, p, j, m)``
       - ``C::vector_type``
       - Vector *v* with values *v*\ `i`:sub: = *m*\ `i`:sub: ? ``p[j[i]]`` : *u*\ `i`:sub:.
 
-    * - ``C::scatter(J{}, u, p, j)``
+    * - ``C::scatter(tag<J>{}, u, p, j)``
       - ``void``
       - Write values *u*\ `i`:sub: to ``p[j[i]]``.
 
-    * - ``C::scatter(J{}, u, p, j, m)``
+    * - ``C::scatter(tag<J>{}, u, p, j, m)``
       - ``void``
       - Write values *u*\ `i`:sub: to ``p[j[i]]`` for lanes *i* where *m*\ `i`:sub: is true.
 
+    * - ``C::compound_indexed_add(tag<J>{}, u, p, j, z)``
+      - ``void``
+      - Update values ``p[j[i]] += u[i]`` for lanes *i*, subject to constraint *z*.
+
+.. rubric:: Casting
+
+Implementations can provide optimized versions of lane-wise
+value casting from other specific implementation classes.
+
+The first argument is a dummy argument
+of type *J*, used only to disambiguate overloads.
+
+.. list-table::
+    :header-rows: 1
+    :widths: 20 20 60
+
+    * - Expression
+      - Type
+      - Description
+
+    * - ``C::cast_from(J{}, p, j)``
+      - ``C::vector_type``
+      - Vector *v* with values *v*\ `i`:sub: = *w*\ `i`:sub:, cast to ``C::scalar_type``.
```

Comment:
The description isn't related to the arguments of cast_from. Also, why isn't J{} passed through tag?

Reply:
It definitely should have a `tag` attached.